websearch/websearch/src/durability.rs

```diff
@@ -103,46 +120,60 @@ mod durable_impl {
                     }
                 });
 
-                let persisted_params = result?;
-                durability.persist_infallible(
-                    StartSearchInput {
-                        params: persisted_params.clone(),
-                    },
-                    persisted_params.clone(),
-                );
-                Ok(SearchSession::new(DurableSearchSession::<Impl>::live(
-                    Impl::unwrapped_search_session(persisted_params).unwrap(),
-                )))
+                match result {
+                    Ok(persisted_params) => {
+                        durability
+                            .persist(persisted_params.clone(), Ok(persisted_params.clone()))?;
```

Comment:
This doesn't make sense currently.

* If making a new search session is a pure (deterministic) function from search params to session, we don't need to make it durable.
* If it is an impure function we should be persisting the constructed session (by serializing it to ReplayState and then reconstructing it from replay state in the replay)

I think the first case applies here, but please double check and adjust accordingly.